crates/storage/src/connection/block.rs

```diff
@@ -327,8 +290,7 @@ pub(super) fn block_header(
     tx: &Transaction<'_>,
     block: BlockId,
 ) -> anyhow::Result<Option<BlockHeader>> {
-    // TODO: is LEFT JOIN reasonable? It's required because version ID can be null for non-existent versions.
-    const BASE_SQL: &str = "SELECT * FROM block_headers LEFT JOIN starknet_versions ON block_headers.version_id = starknet_versions.id";
+    const BASE_SQL: &str = "SELECT * FROM block_headers";
```

Comment:
Minor suggestion: I think now that this expression is so simple lets merge it inside the match below? That way we just have three static str instead of an allocation each time.